.gitlab-ci.yml

```diff
@@ -39,63 +39,65 @@ build-push-portal-image:
   before_script:
   - docker build -f dockerfiles/build/task-runner/Dockerfile -t $DOCKER_IMAGE_TASK_RUNNER .
   - mkdir $PLUGINS_PATH
+  - plugins=$(grep -v -e '^\s*$' -e '^#' ./dockerfiles/build/www/wp-plugins/local.txt)
   - >
-    plugins=$(grep -v -e '^\s*$' -e '^#' ./dockerfiles/build/www/wp-plugins/local.txt);
     for plugin in $plugins;
     do
         curl \
           --header "JOB-TOKEN: $CI_JOB_TOKEN" \
           --location "$CI_API_V4_URL/projects/$CI_PROJECT_ID/packages/generic/wp-plugins/1.0.0/$plugin" \
           --output $PLUGINS_PATH/$plugin;
     done
+  - echo $DOCKER_DEPLOY_TOKEN | docker login -u $DOCKER_DEPLOY_USER $DOCKER_IMAGE_REGISTRY --password-stdin
 
   script:
-  - echo $DOCKER_IMAGE_REGISTRY
-  - echo $DOCKER_IMAGE_PORTAL
   - >
     docker build
     --target production
     --build-arg TASK_RUNNER_IMAGE=$DOCKER_IMAGE_TASK_RUNNER
     --build-arg LOCAL_WP_PLUGINS_DIR=$PLUGINS_PATH
     -t $DOCKER_IMAGE_PORTAL
     .
-  - echo $DOCKER_DEPLOY_TOKEN | docker login -u $DOCKER_DEPLOY_USER $DOCKER_IMAGE_REGISTRY --password-stdin
   - docker image tag $DOCKER_IMAGE_PORTAL $DOCKER_IMAGE_PORTAL:$CI_COMMIT_SHORT_SHA
   - docker push $DOCKER_IMAGE_PORTAL:$CI_COMMIT_SHORT_SHA
 
+  after_script:
+  - docker logout $DOCKER_IMAGE_REGISTRY
+
 tag-image:
   stage: tag
   extends: .validate-deployment-tag
 
-  script:
+  before_script:
   - echo $DOCKER_DEPLOY_TOKEN | docker login -u $DOCKER_DEPLOY_USER $DOCKER_IMAGE_REGISTRY --password-stdin
+
+  script:
   - docker pull $DOCKER_IMAGE_PORTAL:$CI_COMMIT_SHORT_SHA
   - docker image tag $DOCKER_IMAGE_PORTAL:$CI_COMMIT_SHORT_SHA $DOCKER_IMAGE_PORTAL:$DEPLOYMENT_TAG
   - docker push $DOCKER_IMAGE_PORTAL:$DEPLOYMENT_TAG
 
+  after_script:
+  - docker logout $DOCKER_IMAGE_REGISTRY
+
 update-instance:
   stage: deploy
   extends: .validate-deployment-tag
 
   before_script:
-  - >
-    if [ ! -f ~/.ssh/known_hosts ]; then
-      touch ~/.ssh/known_hosts;
-      chmod 644 ~/.ssh/known_hosts;
-    fi
-  - export INSTANCE_HOST="$(ssh-keyscan -t rsa,ed25519 $INSTANCE_IP)"
   - >
     if ! grep -q $INSTANCE_HOST ~/.ssh/known_hosts; then
       $INSTANCE_HOST >> ~/.ssh/known_hosts;
     fi
```

Comment:
This command is not required anymore since we use we don't use the `known_hosts` file (because we use `StrictHostKeyChecking=no`)